code/game/objects/items/airbubble.dm

```diff
@@ -0,0 +1,377 @@
+	icon = 'icons/obj/airbubble.dmi'
+	icon_state = "airbubble"
+	icon_closed = "airbubble"
+	icon_opened = "airbubble_open"
+	open_sound = 'sound/items/zip.ogg'
+	close_sound = 'sound/items/zip.ogg'
+	var/item_path = /obj/item/airbubble
+	var/zipped = 0
+	density = 0
+	storage_capacity = 20
+	var/contains_body = 0
+	var/used = 1
+
+	var/use_internal_tank = 1
+	var/datum/gas_mixture/inside_air
+	var/internal_tank_valve = 45 // arbitrary for now
+	var/obj/item/weapon/tank/internal_tank
+	var/process_ticks = 0
+
+/obj/structure/closet/airbubble/can_open()
+	if(zipped)
+		return 0
+	return 1
+
+/obj/structure/closet/airbubble/can_close()
+	var/turf/T = get_turf(src)
+	for(var/obj/structure/closet/closet in T)
+		if(closet != src)
+			return 0
+	var/mob_num = 0
+	for(var/mob/living/M in T)
+		mob_num += 1
+		if(mob_num > 1)
+			visible_message("<span class='warning'>[src] can only fit one person.</span>")
+			return 0
+	return 1
+
+/obj/structure/closet/airbubble/dump_contents()
+
+	for(var/obj/I in src)
+		if(!istype(I, /obj/item/weapon/tank))
+			I.forceMove(loc)
+
+	for(var/mob/M in src)
+		M.forceMove(loc)
+		if(M.client)
+			M.client.eye = M.client.mob
+			M.client.perspective = MOB_PERSPECTIVE
+
+/obj/structure/closet/airbubble/Initialize()
+	. = ..()
+	add_inside()
+	START_PROCESSING(SSfast_process, src)
+
+/obj/structure/closet/airbubble/Destroy()
+	STOP_PROCESSING(SSfast_process, src)
+	qdel(internal_tank)
+	if(parts)
+		new parts(loc)
+	if (smooth)
+		queue_smooth_neighbors(src)
+	return ..()
+
+/obj/structure/closet/airbubble/close()
+	if(!opened)
+		return 0
+	if(!can_close())
+		return 0
+
+	var/stored_units = 0
+
+	if(store_misc)
+		stored_units += store_misc(stored_units)
+	if(store_items)
+		stored_units += store_items(stored_units)
+	if(store_mobs)
+		stored_units += store_mobs(stored_units)
+
+	icon_state = icon_closed
+	opened = 0
+
+	playsound(loc, close_sound, 25, 0, -3)
+	density = 1
+	add_inside()
+	return 1
+
+/obj/structure/closet/airbubble/MouseDrop(over_object, src_location, over_location)
+	..()
+	if(!zipped)
+		if((over_object == usr && (in_range(src, usr) || usr.contents.Find(src))))
+			if(!ishuman(usr))	return
+			if(opened)	return 0
+			if(contents.len > 1)	return 0
+			visible_message("[usr] folds up the [src.name]")
+			var/obj/item/airbubble/bag = new /obj/item/airbubble(get_turf(src))
+			bag.internal_tank = internal_tank
+			internal_tank.forceMove(bag)
+			internal_tank = null
+			bag.w_class = ITEMSIZE_LARGE
+			bag.icon_state = "airbubble_man_folded"
+			bag.desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. It appears to be poorly hand folded."
+			qdel(src)
+			return
+
+/obj/structure/closet/airbubble/verb/set_internals(mob/user as mob)
+	set src in oview(1)
+	set category = "Object"
+	set name = "Set Internals"
+	if(use_internal_tank)
+		visible_message(
+			"<span class='warning'>[user] set [src] internals.</span>",
```

Comment:
Shouldnt it be is setting / you are trying to set ? -> Since you are doing it right now (before the do_after)